### ISSUE TITLE
set_global_queue issues unexpected TBB warnings

### ISSUE THREAD

**Issue by oleksandr-pavlyk** [IntelPython/dpctl#371]
```python
import dpctl
cpu_d = dpctl.SyclDevice("cpu")
sd = cpu_d.create_sub_devices(partition=(4,4,4))
ctx = dpctl.SyclContext(sd)

q = dpctl.SyclQueue(ctx, sd[0])
```

Then 

```
In [16]: dpctl.set_global_queue(q)
TBB Warning: The number of workers is currently limited to 11. The request for 31 workers is ignored. Further requests for more workers will be silently ignored until the limit changes.
```

**Comment by oleksandr-pavlyk** [comment]
This is not a bug in `dpctl`, nor is it related to sub-devices. Standalone reproducer was created and set to the compiler team for investigation.

**Comment by oleksandr-pavlyk** [comment]
Verified as fixed in `dpctl` built with oneAPI DPC++ 2021.3.
